### PR TITLE
GIVCAMP-228: Add GTM Container.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Adds GTM container on branch and production deploys

# Review By (Date)
- End of sprint

# Criticality
- Normal

# Review Tasks

1. Open up the preview build and inspect the HTML `<head>` 
2. You should not see the GTM container
3. Pull the code down to your local
4. Add `CONTEXT="production"` to your environment variables
5. Run `npm run dev`
6. Load up the home page on local 
7. Validate that the GTM container is added to the head.

### CODE REVIEW

File: components/GAProvider.tsx
Comment (sherakama) [inline components/GAProvider.tsx:1]: This seemed like the right place to put this.

File: next.config.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const nextConfig = {
   },
   env: {
     DEPLOY_PRIME_URL: process.env.DEPLOY_PRIME_URL,
+    CONTEXT: process.env.CONTEXT,
```
Comment (sherakama) [inline next.config.js:17]: Allow the context variable to show up on the front end.

**Comment by yvonnetangsu** [comment]
I noticed something strange, @sherakama - not related to this PR though. I saw that first in your last PR that was merged I believe. This Netlify box - the preview URL previously don't have 500 at the end, not sure why it's automatically added to it now 🤔 

Screenshot 2024-01-19 at 11 36 09 AM

**Comment by sherakama** [comment]
> I noticed something strange, @sherakama - not related to this PR though. I saw that first in your last PR that was merged I believe. This Netlify box - the preview URL previously don't have 500 at the end, not sure why it's automatically added to it now 🤔
> 
> Screenshot 2024-01-19 at 11 36 09 AM

Huh, weird. Good question.

**Review by yvonnetangsu — APPROVED** [review]
Yup just as described. Shows up after adding context locally. Thank you @sherakama ! 🎉 
Screenshot 2024-01-19 at 4 09 06 PM

**Pull request merged**